### ISSUE TITLE
Exception when parsing call of __path__ from namespace package

### ISSUE THREAD

**Issue by MarKett** [PyCQA/astroid#528]
### Steps to reproduce
1. Create a _test.py_ file with the following content:
```
import ns_package
print("__path__:",ns_package.__path__)
```
2. Create a new folder right next to this .py file named _ns_package_. (optionally, insert python files)
3. Run pylint over the _test.py_ file.

### Current behavior
**Exception:**
>   File "D:\Program Files\Python36\lib\site-packages\pylint\lint.py", line 1347, in __init__
>     linter.check(args)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\lint.py", line 768, in check
>     self._do_check(files_or_modules)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\lint.py", line 901, in _do_check
>     self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\lint.py", line 980, in check_astroid_module
>     walker.walk(ast_node)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\utils.py", line 1014, in walk
>     self.walk(child)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\utils.py", line 1014, in walk
>     self.walk(child)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\utils.py", line 1014, in walk
>     self.walk(child)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\utils.py", line 1011, in walk
>     cb(astroid)
>   File "D:\Program Files\Python36\lib\site-packages\pylint\checkers\typecheck.py", line 714, in visit_attribute
>     if not [n for n in owner.getattr(node.attrname)
>   File "D:\Program Files\Python36\lib\site-packages\astroid\scoped_nodes.py", line 484, in getattr
>     result = [self.special_attributes.lookup(name)]
>   File "D:\Program Files\Python36\lib\site-packages\astroid\interpreter\objectmodel.py", line 110, in lookup
>     return getattr(self, "py" + name)
>   File "D:\Program Files\Python36\lib\site-packages\astroid\interpreter\objectmodel.py", line 142, in py__path__
>     path = os.path.dirname(self._instance.file)
>   File "D:\Program Files\Python36\lib\ntpath.py", line 242, in dirname
>     return split(p)[0]
>   File "D:\Program Files\Python36\lib\ntpath.py", line 204, in split
>     p = os.fspath(p)
> TypeError: expected str, bytes or os.PathLike object, not NoneType

### Expected behavior
No errors

### Versions
Python: 3.6.1 (64bit)
PyLint: 1.8.3
Astroid: 1.6.2

**Comment by MarKett** [comment]
Recomended Bugfix in site-packages\astroid\interpreter\objectmodel.py Line: 140

`        path = os.path.dirname(self._instance.file)`
to
```
        try:
            path = os.path.dirname(self._instance.file)
        except TypeError:
            path = self._instance.path
```

**Comment by PCManticore** [comment]
Thanks for the report! Fixed in both master and 1.6 version. I'll do a release of the new 1.6 sometimes this weekend

**Comment by MarKett** [comment]
Thank you for the fast reaction and bugfix!
